module/zfs/zvol.c

```diff
@@ -459,13 +459,14 @@ uzfs_ioc_stats(zfs_cmd_t *zc, nvlist_t *nvl)
 	}
 	(void) mutex_exit(&zvol_list_mutex);
 
-	if (zc->zc_name[0] == '\0') {
-		spa_t *spa = NULL;
-		nvlist_t *tnvl = fnvlist_alloc();
-		mutex_enter(&spa_namespace_lock);
-		while ((spa = spa_next(spa)) != NULL) {
-			nvlist_t *pnvl = fnvlist_alloc();
 
+	spa_t *spa = NULL;
+	nvlist_t *tnvl = fnvlist_alloc();
+	int flag = 0;
+	mutex_enter(&spa_namespace_lock);
```

Comment:
this is heavy lock.. it can't be taken here

Reply:
ok